runtime/registry.go

```diff
@@ -561,11 +561,25 @@ func (r *registryCache) updateProjectConfig(iwc *instanceWithController) error {
 		}
 		return err
 	}
-	dotEnv, err := rillv1.ParseDotEnv(iwc.ctx, repo, iwc.instanceID)
+
+	instance, err := r.get(iwc.instanceID)
+	if err != nil {
+		return err
+	}
+
+	p, err := rillv1.Parse(iwc.ctx, repo, iwc.instanceID, instance.Environment, instance.OLAPConnector)
```

Comment:
There's a call to `ParseRillYAML` right above that I think is also redundant now that it's anyway doing a full parse. Should be able to just check if `rill.yaml` is missing by accessing `p`.

Reply:
Yes, it is redundant. I also simplified a definition of `UpdateInstanceWithRillYAML` that is called next